File: smallestdivisor.swift
#!/usr/bin/env swift

func square(n: Int) -> Int {
  return n * n
}

func divides(x: Int, y: Int) -> Bool {
  return y % x == 0
}

func smallestDivisor(n: Int) -> Int {
  return findDivisor(n,2)
}

func findDivisor(n: Int, t: Int) -> Int {
  if square(t) > n {
    return n
  } else if divides(t, y: n) {
    return t
  } else {
    return findDivisor(n, t: (t+1))
  }
}

func isPrime(n: Int) -> Bool {
  return smallestDivisor(n) == n
}

for i in 1..<50 {
  let x = isPrime(i)
  print("\(i) is prime: \(x)")
}

File: smallest_divisor.py
#!/usr/bin/env python

def square(n):
  return n * n

def smallest_divisor(n):
  return find_divisor(n,2)

def find_divisor(n,test_divisor):
  if square(test_divisor) > n:
    return n
  elif divides(test_divisor,n):
    return test_divisor
  else:
    return find_divisor(n,(test_divisor+1))

def divides(x,y):
  return y % x == 0

def prime(n):
  return smallest_divisor(n) == n

for i in xrange(1,50):
  if prime(i):
    print str(i) + " is prime"